File: java/DataFrame/src/main/java/coyote/dataframe/ByteArrayType.java
/**
 * 
 */
package coyote.dataframe;

/**
 * @author Steve.Cote
 *
 */
public class ByteArrayType implements FieldType
{
  /** negative size indicates a variable length value is to be expected. */
  private static final int _size = -1;

  private final static String _name = "BYTE";




  public boolean checkType( Object obj )
  {
    return ( obj instanceof byte[] );
  }




  public byte[] encode( Object obj )
  {
    return (byte[])obj;
  }




  public Object decode( byte[] value )
  {
    return value;
  }




  public String getTypeName()
  {
    return _name;
  }




  public boolean isNumeric()
  {
    return false;
  }




  public int getSize()
  {
    return _size;
  }

}

File: csharp/DataFrame/Coyote/DataFrame/ByteArrayType.cs
#region license
// Copyright (c) 2006 Stephan D. Cote' - All rights reserved.
// 
// This program and the accompanying materials are made available under the 
// terms of the MIT License which accompanies this distribution, and is 
// available at http://creativecommons.org/licenses/MIT/
// *
// Contributors:
//   Stephan D. Cote 
//      - Initial API and implementation
#endregion
using System;
using System.Text;

namespace Coyote.DataFrame {


    public class ByteArrayType : FieldType {

        /// negative size indicates a variable length value is to be expected. 
        private const int _size = -1;

        private const string _name = "BYTE";




        public virtual bool CheckType( object obj ) {
            return (obj is sbyte[]);
        }




        public virtual byte[] Encode( object obj ) {
            return (byte[])obj;
        }




        public virtual object Decode( byte[] @value ) {
            return @value;
        }




        public virtual string TypeName {
            get {
                return _name;
            }
        }




        public virtual bool IsNumeric {
            get {
                return false;
            }
        }




        public virtual int Size {
            get {
                return _size;
            }
        }




        public virtual string StringValue( byte[] val ) {
            object obj = Decode( val );
            if ( obj != null )
                return obj.ToString();
            else
                return "";
        }

    }

}